pyro/distributions/transforms/block_naf.py

```diff
@@ -152,7 +155,12 @@ def log_abs_det_jacobian(self, x, y):
         """
         Calculates the elementwise determinant of the log jacobian
         """
-        return self._cached_logDetJ.sum(-1)
+        x_old, y_old = self._cached_x_y
```

Comment:
isn't it a bit strange to put this check here but not in other transformed distributions that have caching? either all should be fixed or none? and if all, maybe it can be done in some more uniform way?

Reply:
I agree, it makes sense to uniformly handle this for all transforms that cache the log det term. I'm guessing all of them are in Pyro since torch transforms don't cache this term. We could add a method to `TransformModule` to handle this instead, if that's the case. Please feel free to defer this to a later PR.

Reply:
I'll leave this as is for now